### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Beelzebu**
- Added RedisBungee #63 #77 
- Cache is now refreshed after any command #109

### CODE REVIEW

File: src/main/java/me/leoko/advancedban/manager/CommandManager.java [hunk 1]
```diff
@@ -117,19 +120,19 @@ public void onCommand(final Object sender, final String cmd, final String[] args
...
-                                    (pt.getBasic() == PunishmentType.BAN && PunishmentManager.get().isBanned(uuid))) {
+                            if ((pt.getBasic() == PunishmentType.MUTE && PunishmentManager.get().isMuted(uuid))
+                                    || (pt.getBasic() == PunishmentType.BAN && PunishmentManager.get().isBanned(uuid))) {
                                 MessageManager.sendMessage(sender, pt.getBasic().getConfSection() + ".AlreadyDone", true, "NAME", args[0]);
                                 return;
                             }
 
-                            new Punishment(name, uuid, reason != null ? reason.toString() : null, mi.getName(sender), isTemp && end == -1 ? PunishmentType.BAN : pt, TimeManager.getTime(), end, isTemp && args[1].matches("#.+") ? args[1].substring(1) : null, -1).create();
+                            new Punishment(name, uuid, reason != null ? reason.toString() : "none", mi.getName(sender), isTemp && end == -1 ? PunishmentType.BAN : pt, TimeManager.getTime(), end, isTemp && args[1].matches("#.+") ? args[1].substring(1) : null, -1).create();
```
Comment (DevLeoko) [inline src/main/java/me/leoko/advancedban/manager/CommandManager.java]: The reason has to be null as the getReason() method checks if it's null and if so returns the default reason:
https://github.com/Beelzebu/AdvancedBan/blob/1f18591de0d3617fa63d596d7a6330c73d3c2981/src/main/java/me/leoko/advancedban/utils/Punishment.java#L50

File: src/main/java/me/leoko/advancedban/manager/CommandManager.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package me.leoko.advancedban.manager;
 
+import com.imaginarycode.minecraft.redisbungee.RedisBungee;
```
Comment (DevLeoko) [inline src/main/java/me/leoko/advancedban/manager/CommandManager.java]: Not sure about that but I think this is gonna give a NoClassDefFoundError if the server does not use the RedisBungee as the Scope for this API is set to "provided" in the POM.
Same for all other imports of com.imaginarycode.minecraft.* except for the ones in PubSubMessageListener.java

Comment (Beelzebu) [inline src/main/java/me/leoko/advancedban/manager/CommandManager.java]: I tested I without redis and there is no error, the code of redis is only called when is detected if no, is ignored, if is called and not detected it gonna give a NoClassDefFoundError.
The only error that I have testing without plugins is an error with bStats
`java.lang.IllegalAccessError: tried to access class me.leoko.advancedban.bungeecord.Metrics$1 from class me.leoko.advancedban.bungee.Metrics
    at me.leoko.advancedban.bungee.Metrics.startSubmitting(Metrics.java:167)
    at me.leoko.advancedban.bungee.Metrics.<init>(Metrics.java:102)
    at me.leoko.advancedban.bungee.BungeeMethods.setupMetrics(BungeeMethods.java:131)
    at me.leoko.advancedban.Universal.setup(Universal.java:48)
    at me.leoko.advancedban.bungee.BungeeMain.onEnable(BungeeMain.java:22)
    at net.md_5.bungee.api.plugin.PluginManager.enablePlugins(PluginManager.java:226)
    at net.md_5.bungee.BungeeCord.start(BungeeCord.java:282)
    at net.md_5.bungee.BungeeCordLauncher.main(BungeeCordLauncher.java:55)
    at net.md_5.bungee.Bootstrap.main(Bootstrap.java:15)
`
Seems that other plugin was providing it before.

**Comment by DevLeoko** [comment]
Thank you very much that you help with the development :D
If you are planning to do contribute any further to this project feel free to add me on Skype (Leoko33) or Discord (Leoko#4115) then I can answer you any questions about the existing code and we could agree on who codes which feature so that we don't work on the same ones ;) If not that's fine, thank you anyway ^^

**Comment by Beelzebu** [comment]
I  want to thank you for making this amazing free software plugin and  demonstrate that the minecraft community can create wonderful things.
I would like to continue working on support for redisbungee as it is one of the few things missing in this plugin to be universal at least with spigot and bungeecord.
I don't use Skype, sometimes I use Discord but I prefer TeamSpeak3, I sent you an request in discord anyways. Although the place where I am most present is in spigot, my username is Broxxx.

**Pull request merged**